clients/src/main/java/org/apache/kafka/common/requests/ElectLeadersRequest.java

```diff
@@ -0,0 +1,131 @@
+        }
+
+        @Override
+        public String toString() {
+            return "ElectLeadersRequest("
+                + "electionType=" + electionType
+                + ", topicPartitions=" + ((topicPartitions == null) ? "null" : MessageUtil.deepToString(topicPartitions.iterator()))
+                + ", timeoutMs=" + timeoutMs
+                + ")";
+        }
+
+        private ElectLeadersRequestData toRequestData(short version) {
+            if (electionType != ElectionType.PREFERRED && version == 0) {
+                throw new UnsupportedVersionException("API Version 0 only supports PREFERRED election type");
+            }
+
+            ElectLeadersRequestData data = new ElectLeadersRequestData()
+                .setTimeoutMs(timeoutMs);
+
+            if (topicPartitions != null) {
+                for (Map.Entry<String, List<Integer>> tp : CollectionUtils.groupPartitionsByTopic(topicPartitions).entrySet()) {
+                    data.topicPartitions().add(new ElectLeadersRequestData.TopicPartitions().setTopic(tp.getKey()).setPartitionId(tp.getValue()));
+                }
+            } else {
+                data.setTopicPartitions(null);
+            }
+
+            data.setElectionType(electionType.value);
+
+            return data;
+        }
+    }
+
+    private final ElectLeadersRequestData data;
+    private final short version;
+
+    private ElectLeadersRequest(ElectLeadersRequestData data, short version) {
+        super(ApiKeys.ELECT_LEADERS, version);
+        this.data = data;
+        this.version = version;
+    }
+
+    public ElectLeadersRequest(Struct struct, short version) {
+        super(ApiKeys.ELECT_LEADERS, version);
+        this.data = new ElectLeadersRequestData(struct, version);
+        this.version = version;
+    }
+
+    public ElectLeadersRequestData data() {
+        return data;
+    }
+
+    @Override
+    public AbstractResponse getErrorResponse(int throttleTimeMs, Throwable e) {
+        ApiError apiError = ApiError.fromThrowable(e);
+        Map<String, Map<Integer, ApiError>> electionResult = new HashMap<>();
+        if (version == 0) {
+            for (TopicPartitions topic : data.topicPartitions()) {
+                Map<Integer, ApiError> partitionResult = new HashMap<>();
+                for (Integer partitionId : topic.partitionId()) {
+                    partitionResult.put(partitionId, apiError);
+                }
+                electionResult.put(topic.topic(), partitionResult);
+            }
+        }
+
+        return new ElectLeadersResponse(throttleTimeMs, apiError.error().code(), electionResult, version);
+    }
+
+    public static ElectLeadersRequest parse(ByteBuffer buffer, short version) {
+        return new ElectLeadersRequest(ApiKeys.ELECT_LEADERS.parseRequest(version, buffer), version);
+    }
+
+    /**
+     * Visible for testing.
+     */
+    @Override
+    public Struct toStruct() {
```

Comment:
Looks like we can make this `protected` like the parent class and remove the "Visible for testing" comment.